pex/pex.py

```diff
@@ -656,20 +657,53 @@ def execute_interpreter(self):
                 sys.argv = args
                 return self.execute_content(arg, content)
         else:
-            if self._vars.PEX_INTERPRETER_HISTORY:
-                import atexit
+            try:
                 import readline
-
-                histfile = os.path.expanduser(self._vars.PEX_INTERPRETER_HISTORY_FILE)
-                try:
-                    readline.read_history_file(histfile)
-                    readline.set_history_length(1000)
-                except OSError as e:
-                    sys.stderr.write(
-                        "Failed to read history file at {} due to: {}".format(histfile, e)
+            except ImportError:
+                if self._vars.PEX_INTERPRETER_HISTORY:
+                    pex_warnings.warn(
+                        "PEX_INTERPRETER_HISTORY was requested which requires the `readline` "
+                        "module, but the current interpreter at {python} does not have readline "
+                        "support.".format(python=sys.executable)
                     )
+            else:
```

Comment:
`python` has `-S` that (among other things) disables completion. It looks like this is unconditional, if the modules exist.

Is there potential risks/downsides to doing this unconditionally without a knob to turn it off, beyond the (seemingly small) performance cost of this set-up?

I assume that it's harmless enough, just double checking.

Reply:
Yeah, I got familiar with the CPython site.py code that -S disables in order to get this all working. I can't see any harm.